axum/src/routing/path_router.rs

```diff
@@ -79,6 +82,18 @@ where
         Ok(())
     }
 
+    pub(super) fn method_not_allowed_fallback<H, T>(&mut self, handler: H)
+    where
+        H: Handler<T, S>,
+        T: 'static,
+    {
+        for (_, endpoint) in self.routes.iter_mut() {
+            if let Endpoint::MethodRouter(rt) = endpoint {
+                *rt = rt.clone().fallback(handler.clone());
```

Comment:
I suggested in #2251 to not use the existing `MethodRouter::fallback` but to have a new method that doesn't overwrite an existing fallback, for cases like this:

```rust
let app = Router::new()
    .route("/", get(index).fallback(root_fallback))
    .route("/foo", /* foo MethodRouter */)
    .route("/bar", /* bar MethodRouter */)
    .route("/baz", /* baz MethodRouter */)
    .method_not_allowed_fallback(resource_mna_fallback);
```

I would argue that people are going to expect `root_fallback` to apply to a `POST /` request, whereas as far as I can tell, this implementation would call `resource_mna_fallback` instead.

Reply:
Ahhhh okay. I actually misunderstood you back then. But with this review, I think i get it. I will fix it in the coming days when I get back to it. Also agree with you regarding the tests. I will ping you when I think it is ready for re-review. Thank you very much!

Reply:
No need to ping me, I get notified when you push changes.